elf_binary.cc

```diff
@@ -295,20 +338,23 @@ Elf_Addr ELFBinary::OffsetFromAddr(Elf_Addr addr) {
 
 std::string ELFBinary::ShowDynSymtab() {
     LOGF("ShowDynSymtab\n");
-    std::vector<std::string> res(syms_.size() + 1);
+    std::stringstream ss;
     for (auto it : syms_) {
-        std::stringstream ss;
-        ss << it.first.second << ": " << it.first.first << " ";
-
-        if (versym_) {
-            ss << ShowVersym(it.first.second) << "\n";
+        ss << it.name << ": ";
+
+        if (it.ver == -1) {
```

Comment:
I feel now there are too much `if (x == VER_NDX_LOCAL) {} else if (x == VER_NDX_GLOBAL) {} else`. Could you consider factoring some of them out to helper functions?

Reply:
Thank you. I do it.